src/components/views/rooms/RoomBreadcrumbs.js

```diff
@@ -35,21 +35,42 @@ export default class RoomBreadcrumbs extends React.Component {
 
     componentWillMount() {
         this._dispatcherRef = dis.register(this.onAction);
+
+        const roomStr = localStorage.getItem("mx_breadcrumb_rooms");
+        if (roomStr) {
+            try {
+                const roomIds = JSON.parse(roomStr);
+                this.setState({
+                    rooms: roomIds.map((r) => {
+                        return {
+                            room: MatrixClientPeg.get().getRoom(r),
+                            animated: false,
+                        };
+                    }).filter((r) => r.room),
+                });
+            } catch (e) {
+                console.error("Failed to parse breadcrumbs:", e);
+            }
+        }
     }
 
     componentWillUnmount() {
         dis.unregister(this._dispatcherRef);
     }
 
-    componentDidUpdate() {
+    componentDidUpdate(props, state) {
```

Comment:
nit: doesn't look like you're using these?

Reply:
I'm using state, but the props get in the way of the ordering.

Reply:
I can see you are using `this.state`, but fail to spot usage of `state`?

Reply:
That would be a bug that I failed to catch ._.

Reply:
... and by that I mean I'll take out the `prop, state` declarations because they aren't doing much